response_operations_ui/views/messages.py

```diff
@@ -111,22 +114,66 @@ def _populate_form_details_from_hidden_fields(form):
     return form
 
 
-@messages_bp.route('/', methods=['GET'])
+@messages_bp.route('/', methods=['GET', 'POST'])
 @login_required
-def view_messages():
-    # Currently the filtering is only being done with parameters.  In the future, the session
-    # will have the a list of survey_ids the user is allowed to see and the parameters for the
-    # backstage call can be populated by looking at the session instead of http parameters.
-    params = {
-        'label': request.args.get('label'),
-        'survey': request.args.get('survey')
-    }
-    breadcrumbs = [{"title": "Messages"}]
+def view_select_survey():
+    breadcrumbs = [{"title": "Messages", "link": "/messages"},
+                   {"title": "Filter by survey"}]
+
+    survey_list = [survey.value for survey in Surveys]
+
+    if request.method == 'POST':
+        selected_survey = request.form.get('radio-answer')
+        if not selected_survey:
+            return render_template("message_select_survey.html",
+                                   breadcrumbs=breadcrumbs,
+                                   selected_survey=None,
+                                   survey_list=survey_list,
+                                   response_error=True)
```

Comment:
We could tidy this up a bit by doing something like this:

    if request.method == 'POST':
        selected_survey = request.form.get('radio-answer')
        if selected_survey:
            return redirect(url_for("messages_bp.view_selected_survey",
                            selected_survey=selected_survey))
        else:
            response_error = True
    else:
        response_error = False

    survey_list = [survey.value for survey in Surveys]
    return render_template("message_select_survey.html",
                               breadcrumbs=breadcrumbs,
                               selected_survey=None,
                               response_error=response_error,
                               survey_list=survey_list)